### ISSUE TITLE
[Bug]: Error on the MigrationStartupTask

### ISSUE THREAD

**Issue by vlaguz** [btcpayserver/btcpayserver#5233]
### What is your BTCPay version?

BTCPayServer - 1.11.1

### How did you deploy BTCPay Server?

I'm running BTCPay Server on a Raspiblitz.

### What happened?

The server stopped working.  In the logs I realized it was using a sqlite db, which is now unsupported, so I had to migrate to Postgres.  But when I try to migrate with the command:
```shell
/home/btcpay/dotnet/dotnet run --no-launch-profile --no-build  -c Release --project "/home/btcpay/btcpayserver/BTCPayServer/BTCPayServer.csproj"  -- --sqlitefile=sqllite.db --postgres=btcpayserver
``` 
I receive the error:
```shell
Unhandled exception. System.ArgumentException: Format of the initialization string does not conform to specification starting at index 0.
``` 
I suppose the Postgres' connection string is not correct, but how do I correct it?

### How did you encounter this bug?

Trying to run the server normally.

### Relevant log output

```shell
btcpay@raspberrypi:~/btcpayserver/BTCPayServer $ /home/btcpay/dotnet/dotnet run --no-launch-profile --no-build  -c Release --project "/home/btcpay/btcpayserver/BTCPayServer/BTCPayServer.csproj"  -- --sqlitefile=sqllite.db --postgres=btcpayserver
info: Configuration:  Data Directory: /home/btcpay/.btcpayserver/Main
info: Configuration:  Configuration File: /home/btcpay/.btcpayserver/Main/settings.config
info: BTCPayServer.Plugins.PluginManager: Loading plugins from /home/btcpay/.btcpayserver/Plugins
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.Shopify - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.PointOfSale - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.PayButton - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.NFC - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.Crowdfund - 1.11.1
info: Configuration:  Supported chains: BTC
info: Configuration:  BTC: Explorer url is http://127.0.0.1:24444/
info: Configuration:  BTC: Cookie file is /home/btcpay/.nbxplorer/Main/.cookie
info: Configuration:  Network: Mainnet
Unhandled exception. System.ArgumentException: Format of the initialization string does not conform to specification starting at index 0.
   at System.Data.Common.DbConnectionOptions.GetKeyValuePair(String connectionString, Int32 currentPosition, StringBuilder buffer, Boolean useOdbcRules, String& keyname, String& keyvalue)
   at System.Data.Common.DbConnectionOptions.ParseInternal(Dictionary`2 parsetable, String connectionString, Boolean buildChain, Dictionary`2 synonyms, Boolean firstKey)
   at System.Data.Common.DbConnectionOptions..ctor(String connectionString, Dictionary`2 synonyms, Boolean useOdbcRules)
   at System.Data.Common.DbConnectionStringBuilder.set_ConnectionString(String value)
   at Npgsql.NpgsqlConnectionStringBuilder..ctor(String connectionString)
   at BTCPayServer.Hosting.ToPostgresMigrationStartupTask.ExecuteAsync(CancellationToken cancellationToken) in /home/btcpay/btcpayserver/BTCPayServer/Hosting/ToPostgresMigrationStartupTask.cs:line 175
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.StartWithTasksAsync(IWebHost webHost, CancellationToken cancellationToken) in /home/btcpay/btcpayserver/BTCPayServer/Extensions/WebHostExtensions.cs:line 19
   at BTCPayServer.Program.Main(String[] args) in /home/btcpay/btcpayserver/BTCPayServer/Program.cs:line 77
```

### What browser do you use?

Not important

### Are you sure this is a bug report?

- [x] I confirm this is a bug report

**Comment by pavlenex** [comment]
@NicolasDorier this one is for you :)

**Comment by dennisreimann** [comment]
@vlaguz The `postgres` argument expects a connection string like this:

```
"User ID=postgres;Include Error Detail=true;Host=127.0.0.1;Port=39372;Database=btcpayserver"
```

Does this work for you?

**Comment by vlaguz** [comment]
Thank you very much, I adapted the parameters for my environment, now I got another error:

```shell
duplicate key value violates unique constraint "PK_StoreRoles"
``` 
```shell
btcpay@raspberrypi:~ $ /home/btcpay/dotnet/dotnet run --no-launch-profile --no-build  -c Release --project "/home/btcpay/btcpayserver/BTCPayServer/BTCPayServer.csproj"  -- --sqlitefile=sqllite.db --postgres="User ID=xxxxx;Include Error Detail=true;Host=127.0.0.1;Port=5432;Database=btcpay;Password=xxxxxx'"
info: Configuration:  Data Directory: /home/btcpay/.btcpayserver/Main
info: Configuration:  Configuration File: /home/btcpay/.btcpayserver/Main/settings.config
info: BTCPayServer.Plugins.PluginManager: Loading plugins from /home/btcpay/.btcpayserver/Plugins
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.Shopify - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.PointOfSale - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.PayButton - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.NFC - 1.11.1
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.Crowdfund - 1.11.1
info: Configuration:  Supported chains: BTC
info: Configuration:  BTC: Explorer url is http://127.0.0.1:24444/
info: Configuration:  BTC: Cookie file is /home/btcpay/.nbxplorer/Main/.cookie
info: Configuration:  Network: Mainnet
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating from SQLite to Postgres...
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: There is a unfinished migration in postgres... dropping all tables
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetRoles
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetRoleClaims
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUsers
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserClaims
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserLogins
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserRoles
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserTokens
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Fido2Credentials
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Files
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Notifications
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: OffchainTransactions
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PairingCodes
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PayjoinLocks
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PlannedTransactions
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Settings
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Stores
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: ApiKeys
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Apps
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: CustodianAccount
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Forms
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Invoices
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AddressInvoices
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: InvoiceEvents
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: InvoiceSearches
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: LightningAddresses
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PairedSINData
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PaymentRequests
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Payments
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PayoutProcessors
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PendingInvoices
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PullPayments
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Payouts
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Refunds
info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: StoreRoles
Unhandled exception. Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Npgsql.PostgresException (0x80004005): 23505: duplicate key value violates unique constraint "PK_StoreRoles"

DETAIL: Key ("Id")=(Guest) already exists.
   at Npgsql.Internal.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|215_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
  Exception data:
    Severity: ERROR
    SqlState: 23505
    MessageText: duplicate key value violates unique constraint "PK_StoreRoles"
    Detail: Key ("Id")=(Guest) already exists.
    SchemaName: public
    TableName: StoreRoles
    ConstraintName: PK_StoreRoles
    File: nbtinsert.c
    Line: 649
    Routine: _bt_check_unique
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at BTCPayServer.Hosting.ToPostgresMigrationStartupTask.ExecuteAsync(CancellationToken cancellationToken) in /home/btcpay/btcpayserver/BTCPayServer/Hosting/ToPostgresMigrationStartupTask.cs:line 261
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.StartWithTasksAsync(IWebHost webHost, CancellationToken cancellationToken) in /home/btcpay/btcpayserver/BTCPayServer/Extensions/WebHostExtensions.cs:line 19
   at BTCPayServer.Program.Main(String[] args) in /home/btcpay/btcpayserver/BTCPayServer/Program.cs:line 77

``` 
Should I open another issue?

Thanks in advance.

**Comment by NicolasDorier** [comment]
hey @vlaguz it is a bug on our side. Checking this

**Comment by NicolasDorier** [comment]
@vlaguz can you please DM me directly on chat.btcpayserver.org I fear I can't fix it easily and will need to guide you on how to modify your SQLite database to make it migratable.

**Comment by NicolasDorier** [comment]
@vlaguz I actually know where it comes from. It's a bug on our side, will release new version fixing that.

**Comment by NicolasDorier** [comment]
fixed 1.11.4

**Comment by dme1sc** [comment]
A related migration error is happening from 1.11 to 1.13.  Should I open a new issue?
